### PR TITLE
DOC-3335 STO doc :: load SSL certificates into a Docker container at build time

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 1]
```diff
@@ -8,6 +8,58 @@ Harness maintains its own set of scan images for [STO-supported scanners](/docs/
...
+
+:::note
+This workflow is not supported for container image scans.
+:::
+
+1. Save a copy of the following Dockerfile into a folder along with the certificates you want to copy to the image.
+
+2. Update the `FROM`, `COPY`, and `USER` commands as described in the Dockerfile comments.
+
+3. Build the new image and then upload it to your private registry.
```
Comment (GabeAboy) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:23]:
```suggestion
3. Build the new image and then publish it to your private registry.
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 2]
```diff
@@ -8,6 +8,58 @@ Harness maintains its own set of scan images for [STO-supported scanners](/docs/
...
+
+4. Update the scan step in your pipeline as follows:
+
+   1. Update the **Image** setting to point to the new image in your registry.
+   2. If you specified a `USER`in your Dockerfile, set the **Run as User** (`runAsUser`) setting to the user you specified in your Dockerfile.
```
Comment (GabeAboy) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:28]:
```suggestion
   2. If you specified a `USER` in your Dockerfile, set the **Run as User** (`runAsUser`) setting to the user you specified in your Dockerfile.
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 3]
```diff
@@ -8,6 +8,58 @@ Harness maintains its own set of scan images for [STO-supported scanners](/docs/
...
+USER root
+
+# STEP 2 
+# Copy your certificates to the engine
+# You can copy multiple ca from completely different paths into SHARE_CA_PATH
+COPY ./CERTIFICATE_1.pem ../another-folder/CERTIFICATE_2.pem /shared/customer_artifacts/certificates/
+
+
+# FYI establishes trust for certificates in Python and the OS
+RUN sto_plugin --trust-certs
```
Comment (GabeAboy) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:51]:
```suggestion
RUN sto_plugin --trust-certs && sh /bin/setup.sh /usr/local/share/ca-certificates/
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 4]
```diff
@@ -8,6 +8,58 @@ Harness maintains its own set of scan images for [STO-supported scanners](/docs/
...
+USER root
+
+# STEP 2 
+# Copy your certificates to the engine
+# You can copy multiple ca from completely different paths into SHARE_CA_PATH
+COPY ./CERTIFICATE_1.pem ../another-folder/CERTIFICATE_2.pem /shared/customer_artifacts/certificates/
+
+
+# FYI establishes trust for certificates in Python and the OS
+RUN sto_plugin --trust-certs && sh /bin/setup.sh /usr/local/share/ca-certificates/
```
Comment (GabeAboy) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:51]:
```suggestion
RUN sto_plugin --trust-certs && sh /bin/setup.sh
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 5]
```diff
@@ -8,6 +8,58 @@ Harness maintains its own set of scan images for [STO-supported scanners](/docs/
...
+# FYI Root access is required to load and trust certificates
+USER root
+
+# STEP 2 
+# Copy your certificates to the engine
+# You can copy multiple ca from completely different paths into SHARE_CA_PATH
+COPY ./CERTIFICATE_1.pem ../another-folder/CERTIFICATE_2.pem /shared/customer_artifacts/certificates/
+
+
+# FYI establishes trust for certificates in Python and the OS
```
Comment (GabeAboy) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:50]:
```suggestion
# FYI establishes trust for certificates in Python and the OS 
RUN sto_plugin --trust-certs
# Optional: To trust certificates for Java for tools such as
# - Black Duck Hub
# - Checkmarx
# - Sonarqube
# - Veracode
# - NexusIQ
# RUN sh /bin/setup.sh 
```

File: docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md [hunk 6]
```diff
@@ -8,6 +8,65 @@ Harness maintains its own set of scan images for [STO-supported scanners](/docs/
...
+2. Update the `FROM`, `COPY`, and `USER` commands as described in the Dockerfile comments.
+
+3. Build the new image and then publish it to your private registry.
+
+4. Update the scan step in your pipeline as follows:
+
+   1. Update the **Image** setting to point to the new image in your registry.
+   2. If you specified a `USER` in your Dockerfile, set the **Run as User** (`runAsUser`) setting to the user you specified in your Dockerfile.
+
+<!-- Need to finalize where to copy the certs.../shared/customer_artifacts/certificates/ is the STO default -->
```
Comment (aimurphy) [inline docs/security-testing-orchestration/use-sto/set-up-sto-pipelines/download-images-from-private-registry.md:30]: This comment implies some info is missing? Is it ok to publish anyways?

**Review by aimurphy — APPROVED** [review]
lgtm just one comment.

**Pull request merged**